python/rmm/rmm/tests/test_rmm.py

```diff
@@ -284,6 +284,15 @@ def test_rmm_device_buffer_pickle_roundtrip(hb):
     assert hb3 == hb
 
 
+@pytest.mark.parametrize(
+    "managed, pool", list(product([False, True], [False, True]))
+)
+def test_rmm_device_buffer_prefetch(pool, managed):
+    rmm.reinitialize(pool_allocator=pool, managed_memory=managed)
+    db = rmm.DeviceBuffer.to_device(np.zeros(256, dtype="u1"))
+    db.prefetch()  # just test that it doesn't throw
```

Comment:
You might be able to test that a prefetch call was issued with [`cudaMemRangeGetAttribute (void* data, size_t dataSize, cudaMemRangeAttribute attribute, const void* devPtr, size_t count)`, with attribute `cudaMemRangeAttributeLastPrefetchLocation`](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__MEMORY.html#group__CUDART__MEMORY_1g8048f6ea5ad77917444567656c140c5a). It ought to be possible to call this via `cuda-python` ([API docs](https://nvidia.github.io/cuda-python/module/cuda.html#cuda.cuda.cuMemRangeGetAttribute)).

Also:
> Note that this simply returns the last location that the applicaton requested to prefetch the memory range to. It gives no indication as to whether the prefetch operation to that location has completed or even begun.

You wouldn't know if the prefetch completed or not, but you could verify that the prefetch request was issued.

Reply:
...of course, as soon as I scrolled down, I see you did this exact thing in the C++ tests, at Vyas's request. It would be nice to have a corresponding Python API test, since it should be quick to write with cuda-python.

Reply:
OK...

Reply:
CUDA Python results in very ugly code due to its non-pythonic error handling. But I've done what you asked...